ab_tool/static/ab_tool/js/experiment.js

```diff
@@ -14,28 +17,20 @@ angular.module('ABToolExperiment', []).controller(
          * (the fields in the new track form are bound to the variables
          * $scope.newTrackName and $scope.newTrackWeighting)
          */
-        // Do nothing if newTrackName is empty
-        if (!$scope.newTrackName) {
-            return;
-        }
-        
-        var i, len;
-        // Don't add new track if name is the same as an existing track
-        for (i = 0, len = $scope.experiment.tracks.length; i < len; i++) {
-            if ($scope.experiment.tracks[i]["name"] == $scope.newTrackName) {
-                $window.alert("There is already a track with that name");
-                return;
-            }
-        }
-        
         // Don't include value from hidden weight field if uniformRandom is true
         if ($scope.experiment.uniformRandom) {
             $scope.newTrackWeighting = null;
         }
+        var newTrack = {id: null, name: $scope.newTrackName, editing: false,
```

Comment:
I think you could get away with just specifying `name` here and referring to it in conditional below.